### PR TITLE
Packaging Practices: Use go-task for pkgconfig dep search

### PR DESCRIPTION
**Pull request by Staudey**
## Description

This documents the go-task functionality for pkgconfig dep search instead of the old script way.

Plus some minor clean-ups

### CODE REVIEW

File: docs/packaging/packaging-practices.md [hunk 1]
```diff
@@ -172,7 +164,7 @@ pkgconfig(gtk+-wayland-3.0) pkgconfig(gtk+-x11-3.0)
 
 ### Using pkgconfig dependencies
 
-In the `builddeps` list, simply use the `pkgconfig(name)` syntax. For example, to add gtk+-3.0 to the build dependencies, we would do the following:
+In the `builddeps` list, simply use the `pkgconfig(name)` syntax. For example, to add `gtk+-3.0` to the build dependencies, you would do the following:
```
Comment (davidjharder) [inline docs/packaging/packaging-practices.md:167]:
```suggestion
In the `builddeps` list, use the `pkgconfig(name)` syntax. For example, to add `gtk+-3.0` to the build dependencies, you would do the following:
```

"Simply" used this way grinds my gears. Delete on sight

File: docs/packaging/packaging-practices.md [hunk 2]
```diff
@@ -286,7 +277,7 @@ security/cve-xxxx-xxxx.patch
 fix-silliness.patch
 ```
 
-Both of the files above will be applied using `-p1`. If you need to use stripping num, like `-p4`, you can do something like:
+Both of the files above will be applied using `-p1`. If you need to use a different stripping num, like `-p4`, you can do something like:
```
Comment (davidjharder) [inline docs/packaging/packaging-practices.md:280]:
```suggestion
Both of the files above will be applied using stripping number `-p1`. If you need to use a different stripping number, like `-p4`, you can write:
```

**Review by davidjharder — CHANGES_REQUESTED** [review]
Thanks for updating to the `go-task` workflow

I added some notes for more formatting nits

File: docs/packaging/packaging-practices.md [hunk 3]
```diff
@@ -130,29 +128,23 @@ We prefer the use of `pkgconfig` dependencies. Most modern software will use the
...
-A secondary advantage is that this allows for easily switching or replacing a providing package. When no `pkgconfig` name is available (some packages do not provide these, or it doesn’t make sense for them to), you
-may use the explicit package name. Always ensure you select the correct package, i.e. the `-devel` subpackage. This provides the necessary symlinks and headers to build packages.
+A secondary advantage is that this allows for easily switching or replacing a providing package. When no `pkgconfig` name is available (some packages do not provide these, or it doesn’t make sense for them to), you may use the explicit package name. Always ensure you select the correct package, i.e. the `-devel` subpackage. This provides the necessary symlinks and headers to build packages.
 
-### Finding what package provides a dependency (if any)
+### Finding what package provides a pkgconfig dependency (if any)
 
-The script `common/Scripts/epcsearch.py` is used to find packages that satisfy build dependencies. We recommend making an alias for your shell for this.
+You can use go-task to find packages that satisfy `pkgconfig` dependencies. It can search for multiple dependencies at once.
```
[the remark this answers is not in the document]
Comment (Staudey) [reply docs/packaging/packaging-practices.md]: Duh, thanks and done!

File: docs/packaging/packaging-practices.md [hunk 4]
```diff
@@ -130,29 +128,23 @@ We prefer the use of `pkgconfig` dependencies. Most modern software will use the
...
 
 ```bash
-epcsearch gtk+-3.0
-pkgconfig(gtk+-3.0) found in: libgtk-3-devel
-pkgconfig32(gtk+-3.0) found in: libgtk-3-32bit-devel
+pkgconfig(Qt5Core) found in: qt5-base-devel
+pkgconfig(Qt6Core) found in: qt6-base-devel
 ```
 
 You can also determine if there are pkgconfigs available from a -devel package by doing `eopkg info (name)` and looking for the `Provides` key.
```
[the remark this answers is not in the document]
Comment (Staudey) [reply docs/packaging/packaging-practices.md]: Same here

File: docs/packaging/packaging-practices.md (hunk 1 above)
Comment (Staudey) [reply docs/packaging/packaging-practices.md]: Good point

File: docs/packaging/packaging-practices.md (hunk 2 above)
Comment (Staudey) [reply docs/packaging/packaging-practices.md]: Yeah, I wasn't sure about this one, but kept it close to the old version in the end. But this is more readable for sure. Done.

**Review by davidjharder — APPROVED** [review]
Looks good, thanks!

**Pull request merged**